roles/default/kiali-deploy/tasks/main.yml

```diff
@@ -680,6 +680,19 @@
   # this regex is not 100% accurate, but we want to at least catch obvious errors
   - kiali_vars.api.namespaces.label_selector is not regex('^[a-zA-Z0-9/_.-]+=[a-zA-Z0-9_.-]+$')
 
+# If the signing key is not empty string, and is not of the special value secret:name:key,
+# do some validation on it's length
+- name: Validate signing key, if it is set in the CR
+  fail:
+    msg: "Signing key must be 16, 24 or 32 byte length"
+  when:
+    - kiali_vars.auth.strategy != 'anonymous'
+    - kiali_vars.login_token.signing_key != ""
+    - not(kiali_vars.login_token.signing_key | regex_search('secret:.+:.+'))
+    - kiali_vars.login_token.signing_key | length != 16
+    - kiali_vars.login_token.signing_key | length != 24
+    - kiali_vars.login_token.signing_key | length != 32
```

Comment:
would be good to have a simple molecule test for this... somewhere in here? https://github.com/kiali/kiali-operator/blob/master/molecule/config-values-test/converge.yml#L27

also, make sure the other molecule tests pass - like https://github.com/kiali/kiali-operator/blob/master/molecule/null-cr-values-test/converge.yml#L41

Reply:
> also, make sure the other molecule tests pass - like https://github.com/kiali/kiali-operator/blob/master/molecule/null-cr-values-test/converge.yml#L41

That one passes... However, all three auth molecules don't :-)
I'll check what I can do...

> don't need to test all 16, 24 and 32 - just to test to get into there and see this pass through ok

I think I don't get this...

Reply:
> I think I don't get this...

I mean to say: you don't have to test with 3 separate signing keys - if you just test with one (say, of size 16) and see that it works, that's good enough

Reply:
@jmazzitelli I added a simple test. I hope that's enough.